core/modules/admin_bar/admin_bar.module

```diff
@@ -244,6 +244,15 @@ function admin_bar_preprocess_page(&$variables) {
     $settings['position_fixed'] = $setting;
   }
 
+  // Send whether this is an admin page to JS for back to site functionality.
+  $settings['current_path_is_admin'] = FALSE;
+  $settings['current_path'] = current_path();
```

Comment:
This line here causes the "Back to site" link to always point at the **path** of the original page, INSTEAD of the **alias** .

It would be desirable if this functionality used the alias rather than the path. For example if your original path was `example.com/my-page` you want the link to point at that, rather than at, say, `example.com/node/15`.

I will suggest code to do this later today.

Reply:
This is the suggested code (replace lines 249 - 252). This will make the link point at the alias when available (rather than the real path).

```php
  $current_path = current_path();
  if (path_is_admin($current_path)) {
    $settings['current_path_is_admin'] = TRUE;
  }
  $alias = backdrop_get_path_alias($current_path, $language->langcode);
  $settings['current_path'] = $alias;
```

Reply:
Thanks for the suggestion.  I had noticed that but did not know how to fix.

Reply:
@keiserjb, my original comment (which you resolved) still applies here. The Back to site link is not using the alias. You had fixed this, but not have regressed to the original problematic code.